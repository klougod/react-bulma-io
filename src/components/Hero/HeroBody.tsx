import React from 'react'
import PropTypes from 'prop-types'

export const HeroBody = ({ className, children }: any) => (
  <div className={`hero-body ${className}`}>{children}</div>
)

HeroBody.defaultProps = {
  className: ''
}

HeroBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default HeroBody
