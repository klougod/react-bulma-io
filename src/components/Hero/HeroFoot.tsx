import React, { ReactNode } from 'react'

interface HeroFootProps {
  className?: string
  children?: ReactNode
}
export const HeroFoot = ({ className, children }: HeroFootProps) => (
  <div className={`hero-foot ${className}`}>{children}</div>
)

HeroFoot.defaultProps = {
  className: ''
}

export default HeroFoot
