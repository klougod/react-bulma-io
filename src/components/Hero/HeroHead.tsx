import React, { ReactNode } from 'react'

interface HeroHeadProps {
  className?: string
  children?: ReactNode
}
export const HeroHead = ({ className, children }: HeroHeadProps) => (
  <div className={`hero-head ${className}`}>{children}</div>
)

HeroHead.defaultProps = {
  className: ''
}

export default HeroHead
