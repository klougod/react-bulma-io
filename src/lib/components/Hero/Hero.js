import React, { Children } from 'react'
import PropTypes from 'prop-types'

export const Hero = ({ className, children }) => (
  <section className={`hero ${className}`}>
    { Children.toArray(children).filter(c => c.type.name === "HeroHead") }
    { Children.toArray(children).filter(c => c.type.name === "HeroBody") }
    { Children.toArray(children).filter(c => c.type.name === "HeroFoot") }
  </section>
)

Hero.defaultProps = {
  className: ''
}

Hero.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Hero