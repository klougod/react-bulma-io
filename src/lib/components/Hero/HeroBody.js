import React from 'react'
import PropTypes from 'prop-types'

export const HeroBody = ({ className, children }) => (
  <div className={`hero-body ${className}`}>
    { children }
  </div>
)

HeroBody.defaultProps = {
  className: ''
}

HeroBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroBody