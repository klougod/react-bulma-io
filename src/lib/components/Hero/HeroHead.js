import React from 'react'
import PropTypes from 'prop-types'

export const HeroHead = ({ className, children }) => (
  <div className={`hero-head ${className}`}>
    { children }
  </div>
)

HeroHead.defaultProps = {
  className: ''
}

HeroHead.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroHead