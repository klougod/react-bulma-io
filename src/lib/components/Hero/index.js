export { default as Hero } from './Hero'
export { default as HeroBody } from './HeroBody'
export { default as HeroHead } from './HeroHead'
export { default as HeroFoot } from './HeroFoot'
