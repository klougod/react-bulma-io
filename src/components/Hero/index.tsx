import React, { Children, ReactElement, ReactNode } from 'react'

import HeroHead from './HeroHead'
import HeroBody from './HeroBody'
import HeroFoot from './HeroFoot'

const heroHeadType = (<HeroHead />).type
const heroBodyType = (<HeroBody />).type
const heroFootType = (<HeroFoot />).type

interface HeroProps {
  className?: string
  children?: ReactNode
}

export const Hero = ({ className, children }: HeroProps) => (
  <section className={`hero ${className}`}>
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === heroHeadType
    )}
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === heroBodyType
    )}
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === heroFootType
    )}
  </section>
)

Hero.defaultProps = {
  className: ''
}

export default Hero
