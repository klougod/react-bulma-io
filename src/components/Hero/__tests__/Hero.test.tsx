import React from 'react'
import Enzyme from 'enzyme'

import Hero from '../'
import HeroHead from '../HeroHead'
import HeroBody from '../HeroBody'
import HeroFoot from '../HeroFoot'

describe('Hero component test', function () {
  it('should render without crash', function () {
    const heroMock = Enzyme.shallow(<Hero />)
    expect(heroMock).toMatchSnapshot()
  })

  it('should render without crash with correct children', function () {
    const heroMockProps = { className: 'heroClass' }
    const heroMock = Enzyme.shallow(
      <Hero {...heroMockProps}>
        <HeroHead />
        <HeroBody />
        <HeroFoot />
      </Hero>
    )
    expect(heroMock).toMatchSnapshot()
  })
})
