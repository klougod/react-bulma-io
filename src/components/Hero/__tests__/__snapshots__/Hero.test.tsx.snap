// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component test should render without crash 1`] = `
<section
  className="hero "
/>
`;

exports[`Hero component test should render without crash with correct children 1`] = `
<section
  className="hero heroClass"
>
  <Component
    className=""
    key=".0"
  />
  <Component
    className=""
    key=".1"
  />
  <Component
    className=""
    key=".2"
  />
</section>
`;
